### PR TITLE
Documentation for Subaccounts API

### PR DESCRIPTION
**Pull request by tbedford**
## Description

- [x] Overview placeholder for Subaccounts API as requested.
- [x] Moved to be nested under Account API as requested.
- [x] Updated spell check config file
- [x] Add stub for Tutorials
- [x] Add stub for API Reference
- [x] Add documentation constraint `'account/subaccounts'`
- [x] Add ` { 'path' => 'account/subaccounts', 'icon' => 'user', 'icon_colour' => 'blue', 'name' => 'Subaccounts' },` to tutorials controller
- [x] Add `return 'Subaccounts' if product == 'account/subaccounts'` to `tutorial.rb`
- [x] Add redirects for API Reference and Tutorials
- [x] Add Subaccounts to language file
- [x] Add Subaccounts to navigation file   
- [x] Concept added
- [x] Code snippets added
- [x] Minor improvements to Tutorial wording

## Notes

Subaccounts Curl snippets need to be merged first. Then this PR will be updated to point at the latest Curl code snippets before merging.

## Review

A review app is available on Heroku using the usual URL format.

### CODE REVIEW

**Comment by tbedford** [comment]
@mheap: FYI

File: _documentation/account/subaccounts/overview.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+description: The Subaccounts API (Beta) overview.
+---
+
+# Overview of Subaccounts API (Beta)
+
+> **IMPORTANT:** Subaccounts API is released with **restricted availability**.
+
+## Introduction
+
+The Subaccounts API allows to programmatically create and manage subaccounts for separate business units, use cases, product stages, or separate customers. The API empowers developers to handle various aspects of subaccount logistics: manage credit, track usage, set usage limits, suspend subaccounts, and so on.
```
Comment (marklewin) [inline _documentation/account/subaccounts/overview.md]: > The Subaccounts API allows to programmatically create and manage subaccounts

Missing a word between "allows" and "to" here

File: _documentation/account/subaccounts/overview.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+* Transfer balance from the primary account to the subaccount and vice versa
+* List all subaccount balances and credits
+* Show total credit and total balance across all subaccounts
+* Suspend/re-activate the subaccount
+
+## Concepts
+
+### Account balance and credit
+
+All Nexmo accounts have an associated account balance. It shows the amount of funds available for spending. The account balance is deducted each time a chargeable API call is made. When account balance reaches zero, chargeable API calls cannot be executed unless the account has a credit line provided by Nexmo. If an account has a credit, then its account balance can go below zero. A postpaid account is an account that has credit, a prepaid account is an account without credit.
```
Comment (marklewin) [inline _documentation/account/subaccounts/overview.md]: > A postpaid account is an account that has credit, a prepaid account is an account without credit.

Do you mean that a postpaid account has a credit _facility_ and prepaid accounts do not? If so, I think this needs to be mentioned here.

File: _documentation/account/subaccounts/overview.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+Example: A postpaid primary account that initially had a zero balance and was given €100 in credit by Nexmo already used 20€ of the provided credit, that is, its balance was €-20. It had still €80 = |-20 - -100| that it could either spend itself or transfer to one of its subaccounts. The postpaid primary account decided to transfer €20 to Subaccount1. The primary account's balance became €-40 after this operation.
+
+![Balance transfer](/assets/images/subaccounts/balance_transfer.png)
+
+### Credit allocation
+
+A primary account is considered postpaid if it has a credit provided by Nexmo. The postpaid primary account can allocate a part of its credit to one of its subaccounts. Thus, it is possible to have a subaccount with a zero balance but non-zero credit. This subaccount will be able to make API calls until the allocated credit runs out. In general, any account that has a positive balance has `account_balance + |credit_limit|` funds available for spending.
```
Comment (marklewin) [inline _documentation/account/subaccounts/overview.md]: Again, are we talking about a credit _facility_ here? To me, "credit" is just money in the account.

File: _documentation/account/subaccounts/overview.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+Example: In the previous example, subaccount1's balance was 20, subaccount2's balance was 0, and the primary account's balance was -40. The total balance for the prepaid primary account and its subaccounts is: `total_balance = -40 + 20 + 0 = -20`.
+
+## Best practices
+
+* A Nexmo Partner should possess and manage a Nexmo primary account and should create subaccounts for its end customers.
+* The Partner should not use its primary API key (account) to perform API calls. If the Partner wants to use Nexmo API itself, the Partner should create another subaccount.
+* When an end customer is close to using out all of its credit limit the Partner should either allocate additional credit limit to the end customer or wait for the end customer to pay before increasing end customer’s balance, otherwise the end customers API calls will be temporarily blocked.
```
Comment (marklewin) [inline _documentation/account/subaccounts/overview.md]:
```suggestion
* When an end customer is close to reaching its credit limit the Partner ...
```

**Pull request merged**